### PR TITLE
Option to duplicate profile with its permissions

### LINKED ISSUE #16290: Option to duplicate team profiles: suggestion
I would like to have the option to duplicate a team profile in back office. 

This would speed things up :)

### PR DESCRIPTION
**Pull request by kpodemski**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Thanks to this PR, you can easily duplicate employee profiles
| Type?             | new feature
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #16290.
| How to test?      | Follow instructions from Issue
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27966)

### CODE REVIEW

**Review by PierreRambaud — COMMENTED** [review]
Can we wait until the PR about the migrated page is merged?

**Comment by kpodemski** [comment]
@PierreRambaud sure 👍

File: src/Adapter/Profile/CommandHandler/BulkDuplicateProfileHandler.php
Comment (Progi1984) [inline src/Adapter/Profile/CommandHandler/BulkDuplicateProfileHandler.php:27]:
```suggestion
 */

declare(strict_types=1);

namespace PrestaShop\PrestaShop\Adapter\Profile\CommandHandler;
```

File: src/Adapter/Profile/CommandHandler/BulkDuplicateProfileHandler.php
Comment (Progi1984) [inline src/Adapter/Profile/CommandHandler/BulkDuplicateProfileHandler.php:43]:
```suggestion
class BulkDuplicateProfileHandler extends AbstractProfileHandler implements BulkDuplicateProfileHandlerInterface
```

File: src/Adapter/Profile/CommandHandler/BulkDuplicateProfileHandler.php
Comment (Progi1984) [inline src/Adapter/Profile/CommandHandler/BulkDuplicateProfileHandler.php:41]:
```suggestion
 * Class BulkDuplicateProfileHandler
```

File: src/Adapter/Profile/CommandHandler/DuplicateProfileHandler.php
Comment (Progi1984) [inline src/Adapter/Profile/CommandHandler/DuplicateProfileHandler.php:43]:
```suggestion
 */
class DuplicateProfileHandler extends AbstractProfileHandler implements DuplicateProfileHandlerInterface
```

File: src/Core/Domain/Profile/Command/BulkDuplicateProfileCommand.php
Comment (Progi1984) [inline src/Core/Domain/Profile/Command/BulkDuplicateProfileCommand.php:52]:
```suggestion
    public function getProfileIds(): array
```

File: src/Core/Domain/Profile/Command/BulkDuplicateProfileCommand.php
Comment (Progi1984) [inline src/Core/Domain/Profile/Command/BulkDuplicateProfileCommand.php:60]:
```suggestion
    private function setProfileIds(array $profileIds): void
```

File: src/Core/Domain/Profile/Command/BulkDuplicateProfileCommand.php
Comment (Progi1984) [inline src/Core/Domain/Profile/Command/BulkDuplicateProfileCommand.php:27]:
```suggestion
 */

declare(strict_types=1);

namespace PrestaShop\PrestaShop\Core\Domain\Profile\Command;
```

File: src/Core/Domain/Profile/Command/DuplicateProfileCommand.php
Comment (Progi1984) [inline src/Core/Domain/Profile/Command/DuplicateProfileCommand.php:27]:
```suggestion
 */

declare(strict_types=1):

namespace PrestaShop\PrestaShop\Core\Domain\Profile\Command;
```

File: src/Core/Domain/Profile/Command/DuplicateProfileCommand.php
Comment (Progi1984) [inline src/Core/Domain/Profile/Command/DuplicateProfileCommand.php:44]:
```suggestion
    public function __construct(int $profileId)
```

File: src/Core/Domain/Profile/Command/DuplicateProfileCommand.php
Comment (Progi1984) [inline src/Core/Domain/Profile/Command/DuplicateProfileCommand.php:46]:
```suggestion
        $this->profileId = new ProfileId($profileId);
```

File: src/Core/Domain/Profile/Command/DuplicateProfileCommand.php
Comment (Progi1984) [inline src/Core/Domain/Profile/Command/DuplicateProfileCommand.php:52]:
```suggestion
    public function getProfileId(): ProfileId
```

File: src/Core/Domain/Profile/CommandHandler/BulkDuplicateProfileHandlerInterface.php
Comment (Progi1984) [inline src/Core/Domain/Profile/CommandHandler/BulkDuplicateProfileHandlerInterface.php:41]:
```suggestion
    public function handle(BulkDuplicateProfileCommand $command): void;
```

File: src/Core/Domain/Profile/CommandHandler/DuplicateProfileHandlerInterface.php
Comment (Progi1984) [inline src/Core/Domain/Profile/CommandHandler/DuplicateProfileHandlerInterface.php:41]:
```suggestion
    public function handle(DuplicateProfileCommand $command): void;
```

File: src/Core/Domain/Profile/Exception/FailedToDuplicateProfileException.php
Comment (Progi1984) [inline src/Core/Domain/Profile/Exception/FailedToDuplicateProfileException.php:27]:
```suggestion
 */

declare(strict_types=1)

namespace PrestaShop\PrestaShop\Core\Domain\Profile\Exception;
```

File: src/Core/Grid/Definition/Factory/BulkDuplicateActionTrait.php
Comment (Progi1984) [inline src/Core/Grid/Definition/Factory/BulkDuplicateActionTrait.php:27]:
```suggestion
 */

declare(strict_types=1)

namespace PrestaShop\PrestaShop\Core\Grid\Definition\Factory;
```

File: src/Core/Grid/Definition/Factory/BulkDuplicateActionTrait.php
Comment (Progi1984) [inline src/Core/Grid/Definition/Factory/BulkDuplicateActionTrait.php:44]:
```suggestion
    protected function buildBulkDuplicateAction(string $bulkDuplicateRouteName, array $options = []): SubmitBulkAction
```

File: src/Core/Grid/Definition/Factory/BulkDuplicateActionTrait.php
Comment (Progi1984) [inline src/Core/Grid/Definition/Factory/BulkDuplicateActionTrait.php:79]:
```suggestion
    abstract protected function trans(string $id, array $options, string $domain): string;
```

File: src/Core/Grid/Definition/Factory/DuplicateActionTrait.php
Comment (Progi1984) [inline src/Core/Grid/Definition/Factory/DuplicateActionTrait.php:25]:
```suggestion
 */

```

File: src/Core/Grid/Definition/Factory/DuplicateActionTrait.php
Comment (Progi1984) [inline src/Core/Grid/Definition/Factory/DuplicateActionTrait.php:79]:
```suggestion
    abstract protected function trans(string $id, array $options, string $domain): string;
```

File: src/Core/Grid/Definition/Factory/DuplicateActionTrait.php
Comment (Progi1984) [inline src/Core/Grid/Definition/Factory/DuplicateActionTrait.php:46]:
```suggestion
    ): SubmitRowAction {
```

File: src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/ProfileController.php
Comment (Progi1984) [inline src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/ProfileController.php:266]:
```suggestion
    public function duplicateAction(int $profileId): RedirectResponse
```

File: src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/ProfileController.php
Comment (Progi1984) [inline src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/ProfileController.php:321]:
```suggestion
    public function bulkDuplicateAction(Request $request): RedirectResponse
```

**Review by Progi1984 — CHANGES_REQUESTED** [review]
@kpodemski Could you add Behat tests, please ?

**Comment by Julievrz** [comment]
Wording suggestion:

**Modal**

Title: Duplicate selection?
Body: [No body]
Exit buttons: Cancel/Duplicate (Admin.Actions)

(Although I'm wondering if a modal is really necessary here. It's an additional step for the user when there's no big risk. It's not like he/she is about to delete a bunch of items for good.)

**Label**

About the label of the bulk action, I think the whole wording/design could be improved but in the meantime, I suggest keeping the existing string "Duplicate selection" in Admin.Actions, instead of creating the string "Duplicate selected". (It's also more consistent with other pages).

**Comment by kpodemski** [comment]
@PierreRambaud can I continue with this one? :>

**Review by Progi1984 — CHANGES_REQUESTED** [review]
@kpodemski The migrated PR has been merged so you can continue this one (and rebase it).

**Comment by matks** [comment]
@kpodemski do we keep this PR open or do we close? I'm a little sorry but it seems you could not find time to work on it

**Comment by kpodemski** [comment]
@matks 

keep it open, please ;-)

**Comment by kpodemski** [comment]
@Progi1984, would you like me to add a test if the permissions are the same between a source profile and a profile that has been cloned?

**Comment by Progi1984** [comment]
@kpodemski It should be perfect.

**Comment by matks** [comment]
Hi @kpodemski I believe this draft PR is now too stale. I close it for now, just to help navigating old PRs. But we can reopen it anytime you want 😉 obviously. A closed PR does not mean it has to stay closed forever 💪 ! I just do this to make it easier to distinguish old PRs that might be merged one day and PRs that - unfortunately - I think are too old and will never be merged.

Thank you anyway for submitting the PR 👍

**Pull request closed without merging**
